src/library/scala/reflect/ClassTag.scala

```diff
@@ -45,8 +45,9 @@ trait ClassTag[T] extends ClassManifestDeprecatedApis[T] with Equals with Serial
   /** Produces a `ClassTag` that knows how to instantiate an `Array[Array[T]]` */
   def wrap: ClassTag[Array[T]] = ClassTag[Array[T]](arrayClass(runtimeClass))
 
+  // TODO: should be abstract once ClassManifestDeprecatedApis is removed
   /** Produces a new array with element type `T` and length `len` */
-  override def newArray(len: Int): Array[T]
+  override def newArray(len: Int): Array[T] = super.newArray(len)
```

Comment:
Does dotty accept the following definition in trait `ClassTag[T]`?

~~~ scala
def newArray(len: Int): Array[T]
~~~

(abstract, and without the `override` keyword)

Reply:
If I make this one abstract, Dotty enforces that all children of `ClassTag` must implement `newArray` (i.e. `AbstractTypeClassManifest`, `ClassTypeManifest`, `SingletonTypeManifest`, `ClassTypeManifest`, `AbstractTypeManifest`, `WildcardManifest`, `IntersectionTypeManifest`).

This is the Dotty way of preserving the current behavior

Reply:
What would work on the other hand is to remove this definition and just keep the one in `ClassManifestDeprecatedApis`. I kept this one because `ClassManifestDeprecatedApis` was deprecated.

Reply:
Another possible solution is to replace this with the one from `ClassManifestDeprecatedApis` 😄
```diff
diff --git a/src/library/scala/reflect/ClassManifestDeprecatedApis.scala b/src/library/scala/reflect/ClassManifestDeprecatedApis.scala
index 470ba7efc6..a36eef9aea 100644
--- a/src/library/scala/reflect/ClassManifestDeprecatedApis.scala
+++ b/src/library/scala/reflect/ClassManifestDeprecatedApis.scala
@@ -94,9 +94,6 @@ trait ClassManifestDeprecatedApis[T] extends OptManifest[T] {
   def arrayManifest: ClassManifest[Array[T]] =
     ClassManifest.classType[Array[T]](arrayClass[T](runtimeClass), this)
 
-  def newArray(len: Int): Array[T] =
-    java.lang.reflect.Array.newInstance(runtimeClass, len).asInstanceOf[Array[T]]
-
   @deprecated("use wrap.newArray instead", "2.10.0")
   def newArray2(len: Int): Array[Array[T]] =
     java.lang.reflect.Array.newInstance(arrayClass[T](runtimeClass), len)
diff --git a/src/library/scala/reflect/ClassTag.scala b/src/library/scala/reflect/ClassTag.scala
index 313def0c7f..cd02ff7542 100644
--- a/src/library/scala/reflect/ClassTag.scala
+++ b/src/library/scala/reflect/ClassTag.scala
@@ -45,9 +45,11 @@ trait ClassTag[T] extends ClassManifestDeprecatedApis[T] with Equals with Serial
   /** Produces a `ClassTag` that knows how to instantiate an `Array[Array[T]]` */
   def wrap: ClassTag[Array[T]] = ClassTag[Array[T]](arrayClass(runtimeClass))
 
+
   // should be abstract once ClassManifestDeprecatedApis is removed
   /** Produces a new array with element type `T` and length `len` */
-  override def newArray(len: Int): Array[T] = super.newArray(len)
+  def newArray(len: Int): Array[T] =
+    java.lang.reflect.Array.newInstance(runtimeClass, len).asInstanceOf[Array[T]]
 
   /** A ClassTag[T] can serve as an extractor that matches only objects of type T.
    *
```